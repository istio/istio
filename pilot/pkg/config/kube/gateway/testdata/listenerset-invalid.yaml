apiVersion: gateway.networking.k8s.io/v1beta1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: waypoint
  namespace: istio-system
spec:
  addresses:
    - value: istio-ingressgateway
      type: Hostname
  gatewayClassName: istio-waypoint
  listeners:
    - name: mesh
      port: 15008
      protocol: HBONE
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: parent-gateway
  namespace: istio-system
spec:
  allowedListeners:
    namespaces:
      from: All
  addresses:
    - value: istio-ingressgateway
      type: Hostname
  gatewayClassName: istio
  listeners:
    - name: foo
      hostname: foo.com
      protocol: HTTP
      port: 80
---
apiVersion: gateway.networking.x-k8s.io/v1alpha1
kind: XListenerSet
metadata:
  name: port-not-in-service
  namespace: istio-system
spec:
  parentRef:
    name: parent-gateway
    kind: Gateway
    group: gateway.networking.k8s.io
  listeners:
    - name: first
      hostname: first.foo.com
      protocol: HTTP
      port: 12345
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: not-accepted-parent
  namespace: istio-system
spec:
  allowedListeners:
    namespaces:
      from: All
  addresses:
    - value: 0.0.0.0
      type: test.example.com/custom
  gatewayClassName: istio
  listeners:
    - name: foo
      hostname: foo.com
      protocol: HTTP
      port: 80
---
apiVersion: gateway.networking.x-k8s.io/v1alpha1
kind: XListenerSet
metadata:
  name: not-accepted-parent
  namespace: istio-system
spec:
  parentRef:
    name: not-accepted-parent
    kind: Gateway
    group: gateway.networking.k8s.io
  listeners:
    - name: first
      hostname: first.foo.com
      protocol: HTTP
      port: 80
---
apiVersion: gateway.networking.x-k8s.io/v1alpha1
kind: XListenerSet
metadata:
  name: invalid-class
  namespace: istio-system
spec:
  parentRef:
    name: waypoint
    kind: Gateway
    group: gateway.networking.k8s.io
  listeners:
    - name: first
      hostname: first.foo.com
      protocol: HTTP
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: parent-with-no-children
  namespace: istio-system
spec:
  allowedListeners:
    namespaces:
      from: All
  addresses:
    - value: istio-ingressgateway
      type: Hostname
  gatewayClassName: istio
  listeners:
    - name: foo
      hostname: foo.com
      protocol: HTTP
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: parent-no-allowed-listeners
  namespace: istio-system
spec:
  addresses:
    - value: istio-ingressgateway
      type: Hostname
  gatewayClassName: istio
  listeners:
    - name: foo
      hostname: foo.com
      protocol: HTTP
      port: 80
---
apiVersion: gateway.networking.x-k8s.io/v1alpha1
kind: XListenerSet
metadata:
  name: not-allowed
  namespace: istio-system
spec:
  parentRef:
    name: parent-no-allowed-listeners
    kind: Gateway
    group: gateway.networking.k8s.io
  listeners:
    - name: first
      hostname: first.foo.com
      protocol: HTTP
      port: 80