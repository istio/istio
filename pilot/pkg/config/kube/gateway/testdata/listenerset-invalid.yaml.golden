apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/parent-gateway/foo.istio-system
  creationTimestamp: null
  name: parent-gateway-istio-autogenerated-k8s-gateway-foo
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/foo.com
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/parent-no-allowed-listeners/foo.istio-system
  creationTimestamp: null
  name: parent-no-allowed-listeners-istio-autogenerated-k8s-gateway-foo
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/foo.com
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/parent-with-no-children/foo.istio-system
  creationTimestamp: null
  name: parent-with-no-children-istio-autogenerated-k8s-gateway-foo
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/foo.com
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent-namespace: istio-system
    internal.istio.io/parents: XListenerSet/port-not-in-service/first.istio-system
  creationTimestamp: null
  name: port-not-in-service-istio-autogenerated-k8s-gateway-first
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/first.foo.com
    port:
      name: default
      number: 12345
      protocol: HTTP
---
